src/test/java/org/folio/edge/sip2/metrics/MetricsTests.java

```diff
@@ -0,0 +1,86 @@
+package org.folio.edge.sip2.metrics;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import io.micrometer.core.instrument.Timer;
+import org.folio.edge.sip2.parser.Command;
+import org.junit.jupiter.api.Test;
+
+class MetricsTests {
+  @Test
+  void testGetMetrics() {
+    final Metrics m = Metrics.getMetrics(1234);
+    assertNotNull(m);
+  }
+
+  @Test
+  void testSocketError() {
+    final Metrics m = Metrics.getMetrics(1234);
+    m.socketError();
+    m.socketError();
+    assertEquals(2, m.socketErrorCount());
+  }
+
+  @Test
+  void testRequestError() {
+    final Metrics m = Metrics.getMetrics(1234);
+    m.requestError();
+    m.requestError();
+    m.requestError();
+    assertEquals(3, m.requestErrorCount());
+  }
+
+  @Test
+  void testResponseError() {
+    final Metrics m = Metrics.getMetrics(1234);
+    m.responseError();
+    assertEquals(1, m.responseErrorCount());
+  }
+
+  @Test
+  void testScResendError() {
+    final Metrics m = Metrics.getMetrics(1234);
+    m.scResendError();
+    m.scResendError();
+    m.scResendError();
+    m.scResendError();
+    m.scResendError();
+    assertEquals(5, m.scResendErrorCount());
+  }
+
+  @Test
+  void testInvalidMessageError() {
+    final Metrics m = Metrics.getMetrics(1234);
+    m.invalidMessageError();
+    m.invalidMessageError();
+    m.invalidMessageError();
+    m.invalidMessageError();
+    assertEquals(4, m.invalidMessageErrorCount());
+  }
+
+  @Test
+  void testSample() {
+    final Metrics m = Metrics.getMetrics(1234);
+    final Timer.Sample sample = m.sample();
+    assertNotNull(sample);
+  }
+
+  @Test
+  void testCommandTimer() {
+    final Metrics m = Metrics.getMetrics(1234);
+    final Timer.Sample sample = m.sample();
+    final Timer timer = m.commandTimer(Command.UNKNOWN);
+    assertNotNull(timer);
+    final long time = sample.stop(timer);
+    assertTrue(time > 0L);
+  }
+
+  @Test
+  void testStop() {
+    final Metrics m = Metrics.getMetrics(1234);
+    assertNotNull(m);
+    m.stop();
+  }
+}
```

Comment:
Would it be possible to create a semi-unit test illustrating a rate or timer counter of one of the real transactions? The assertion will be time > 0L, for example.  I know we have limitations in our ability to mock out the back end, but I think the SC Status transaction could be used here.

Reply:
Maybe? I'm not sure what we can get when the backend is not specified. I explicitly set `SimpleMeterRegistry` in the case where there is no backend is specified. I'm not sure what Vert.x will do. I'll see if I can whip something up.

Reply:
Actually, I'll create a new story for this type of unit test. I think it will be easier to complete when we have a mock FOLIO server in place (mock integration tests).

Reply:
OK, I was under the impression that getting ACS status positive tests were still working but looking at the tests again they are pretty much disabled. No need to try whipping something up if it's not easy to work with without requiring refactoring.